mlir/lib/Dialect/Math/Transforms/PolynomialApproximation.cpp

```diff
@@ -43,16 +43,16 @@ using namespace mlir::vector;
 struct VectorShape {
   ArrayRef<int64_t> sizes;
   ArrayRef<bool> scalableFlags;
-
-  bool empty() const { return sizes.empty(); }
+  bool scalar = true;
 };
 
 // Returns vector shape if the type is a vector. Returns an empty shape if it is
 // not a vector.
 static VectorShape vectorShape(Type type) {
   auto vectorType = dyn_cast<VectorType>(type);
   return vectorType
-             ? VectorShape{vectorType.getShape(), vectorType.getScalableDims()}
+             ? VectorShape{vectorType.getShape(), vectorType.getScalableDims(),
+                           /*scalar=*/false}
              : VectorShape{};
 }
```

Comment:
I do find `VectorShape` with `scalar` a little confusing (as there's now two empty shape cases). Maybe instead this could return `std::optional<VectorShape>` (and the helpers be updated to take that too)?

Reply:
Removed scalar, changed to std::optional.